### PR TITLE
Force libunwind usage (removes gcc_eh support)

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Force libunwind usage (removes gcc_eh support)

libunwind is reentrant and signal safe, and works faster then then gcc_eh (plus it has some custom patches for problems that have been found during it's usage in ClickHouse).

gcc_eh may be missing in the system (if gcc was not installed), and even if it exists clickhouse uses -nodefaultlibs, so some care should be made to make it work.

Also this library is tiny and there shouln't be any problem to require it always (there is already tendency to require some contrib libraries, i.e. poco).

Cc: @alexey-milovidov (thoughts?)

### CODE REVIEW

**Comment by azat** [comment]
It is not possible, it is still required for darwin

**Comment by azat** [comment]
>It is not possible, it is still required for darwin

Actually libunwind should work with darwin, but stacktraces on darwin are empty, but it is not only in this PR, but the same thing with upstream too

**Comment by alexey-milovidov** [comment]
I'm supportive of this change.

**Comment by alexey-milovidov** [comment]
For some reason, it does not work on RISC-V: https://github.com/ClickHouse/ClickHouse/pull/31398/files#diff-d1333280cb4e5765a4ffbdd3e85525875bd49c0e6c7fd5f67f67001888e09179

We didn't add the RISC-V build to CI, but it works (validated and tested multiple times on various hardware), and we want to add it asap.

**Comment by azat** [comment]
>For some reason, it does not work on RISC-V: [#31398 (files)](https://github.com/ClickHouse/ClickHouse/pull/31398/files#diff-d1333280cb4e5765a4ffbdd3e85525875bd49c0e6c7fd5f67f67001888e09179)

@alexey-milovidov Do you have an error? Or it simply returns empty stacktraces?

**Comment by alexey-milovidov** [comment]
I don't remember, maybe it should work - try it.

**Comment by azat** [comment]
It builds with libunwind, but I'm not able to check does binary works since it simply SIGSEGV in allocator (any binary, the one that you provided in the referenced issue, the one that I compiled w/ and w/o jemalloc), need to dig the root cause, likely it does some allocations with a different allocator (but then, why it SIGSEGV without jemalloc).

**Comment by alexey-milovidov** [comment]
We've successfully tested the RISC-V binary and even included the results into the hardware benchmark: https://benchmark.clickhouse.com/hardware/

File: programs/server/Server.cpp [hunk 1]
```diff
@@ -1650,10 +1650,6 @@ try
 
     /// Describe multiple reasons when query profiler cannot work.
 
-#if !USE_UNWIND
-    LOG_INFO(log, "Query Profiler and TraceCollector are disabled because they cannot work without bundled unwind (stack unwinding) library.");
```
Comment (alexey-milovidov) [inline programs/server/Server.cpp:1654]: Wouldn't it also cause signal-unsafe code on MacOS?

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**